src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/writer/CatalogMultiResourceItemWriter.java

```diff
@@ -0,0 +1,225 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.writer;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.MultiResourceItemWriter;
+import org.springframework.batch.item.file.ResourceAwareItemWriterItemStream;
+import org.springframework.batch.item.file.ResourceSuffixCreator;
+import org.springframework.batch.item.file.SimpleResourceSuffixCreator;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+
+/**
+ * Wraps a {@link ResourceAwareItemWriterItemStream} and creates a new output resource when the count of items written in current resource exceeds
+ * {@link #setItemCountLimitPerResource(int)}. Suffix creation can be customized with {@link #setResourceSuffixCreator(ResourceSuffixCreator)}.
+ * 
+ * Note that new resources are created only at chunk boundaries i.e. the number of items written into one resource is between the limit set by
+ * {@link #setItemCountLimitPerResource(int)} and (limit + chunk size).
+ * 
+ * Code based on original {@link MultiResourceItemWriter} by Robert Kasanicky.
+ * 
+ * @param <T> item type
+ * 
+ * @author Robert Kasanicky
```

Comment:
Who's this guy? If the code is copy/pasted, was the original code GPL compatible (requires to have a licence and it be GPL compatible, source code without a licence is not reusable)